Add containerized retail workflow (#1574)
uyuni-project/uyuni-docs: modules/retail/pages/containerized-saltboot.adoc

[[retail-install-setup-containerized]]
= Set Up the {productname} {smr} Environment using containerized proxy

To set up the {productname} {smr} environment, you will need to have already installed and configured:

* {productname} {smr} Server 4.3 or newer
* one or more {productname} {smr} containerized proxies
* one or more {productname} build host

> This section covers how to configure your  environment using containerized {productname} Proxy for [systemitem]``Saltboot`` deployment.

Reviewer: ```suggestion
The only remaining step to configure is a SaltBoot Group.

This section covers how to configure your environment using containerized {productname} Proxy for [systemitem]``Saltboot Group`` deployment.
```

Author: Added

[NOTE]
====
Containerized {productname} {smr} environment does not use {productname} {smr} Branch Servers.
====

[IMPORTANT]
====
Containerized workflow requires POS images build using {productname} Server 4.3 or newer.
Older images will not work.
====

[IMPORTANT]
====
Containerized workflow not longer implicitly configure DHCP for PXE booting.

> See xref:specialized-guides:salt/salt-formula-dhcpd.adoc[] how to use DHCP formula to configure DHCP server.

Reviewer: We might mention the possible need of DNS formula too.
At least, I required it in my manual setup.

Without using DHCP formula, make sure your DHCP server has correct PXE booting setting pointing to containerized proxy.

> See below for example.

Reviewer: Missing example?

Author: Link added

====

== Saltboot group

Containerized {productname} {smr} is configured in [systemitem]``Saltboot Group``.

> Saltboot Groups are regular server groups with [systemitem]``Saltboot Group`` formula enabled.

Reviewer: ```suggestion
Saltboot Groups are regular system groups with [systemitem]``Saltboot Group`` formula enabled.

The first step is to create a new System Group, from Systems Menu.
This system group must have enabled the Formula "Saltboot Group" from the Formulas tab.
```

Name of the [systemitem]``Saltboot group`` is automatically used as a [systemitem]``branch id``, an identifier for group of machines booted through particular containerized {productname} Proxy.

All [systemitem]``saltboot`` deployed machines though containerized proxy will automatically became members of its [systemitem]``Saltboot group``.

To connect [systemitem]``Saltboot group`` with containerized proxy fill [systemitem]``Image Download Server`` entry with Fully Qualified Domain Name ([literal]``FQDN``) of the containerized proxy.

The rest of configuration is optional.

=== Default boot image

Configure [systemitem]``Default boot image for new registrations`` to specify what boot image should be booted by not yet registered POS terminal.
This is useful when stable boot image is wanted for initial deployments.
Without this setting, newest built boot image is used as default boot image.

If [systemitem]``Default boot image for new registrations`` is set, option to set its version appears under name [systemitem]``Default boot image version`` where specific image version can be set.

=== Kernel option for the saltboot group

> Option [systemitem]``Kernel parameters for the group`` can be used to pass extra kernel options to all POS terminals registered withing this saltboot group.

Reviewer: We should add a comment here to troubleshooting the issue I add, solved by included `id=dhcp`

=== Naming scheme for new registrations

Last three options are related to how will be newly registered machine visible in {productname} Server.

See xref:retail-terminal-names.adoc[] for explanation of possible configurations.

== Comparing containerized and non-containerized workflows

> External DHCP service must be used with containerized Saltboot.

Reviewer: Wow. That could be a big constraint.
And we don't even provide any guide to help the user in this.. let's call it "regression" comparing to what we were offering with the regular retail branch server.

Reviewer: OK, @aaannz clarified that most of our customers are not using the DHCP coming from the Branch server as explained in our Documentation. But instead, they use their own DHCP.
If that's the case for the majority of our customers, then I guess this is not such a big deal.

Containerized workflow relies on updated image building in {productname} Server 4.3 where PXE images are no longer collected as bundle, but kernel, initrd and filesystem image are collected individually.

> Containerized workflow uses new TFTP container which instead of providing files present on the proxy, routes specific TFTP requests as HTTP requests through local squid proxy to {productname} Server.

Reviewer: Do we list which specific requests somewhere?

> Containerized proxy is not a Salt client, it is not possible to call [systemitem]``image-sync`` state.

Reviewer: I would extend this, by explaining how we should synchronize new images now.

Author: Added a note there is no image sync in containerized proxy.

[IMPORTANT]
====
Once POS image is build and made available on {productname} Server, it is immediately available to the Saltboot clients as well.
This may have implications on how images are deployed to production.
====

The following sections differentiate between containerized and regular workflow.
Both are assuming proxy (containerized or in form of {productname} {smr} Branch Server) are available.

Containerized workflow:

. Build POS image
. Configure DHCP server for PXE booting for given network

> . Create [systemitem]``Saltboot group`` and configure it for existing containerized proxy

Reviewer: This section miss the external DHCP, and the need of a system for saltboot formula.

[IMPORTANT]
====
Containerized retail workflow no longer manages DHCP server.
External DHCP service must be used.
====

Containerized workflow relies on updated image building in {productname} Server 4.3 where PXE images are no longer collected as bundle, but kernel, initrd and filesystem image are collected individually.

Containerized workflow uses new TFTP container which instead of providing files present on the proxy, routes specific TFTP requests as HTTP requests through proxy local squid proxy to {productname} Server.

Containerized proxy is not a Salt client, it is not possible to call [systemitem]``image-sync`` state.

> Thanks to these improvements and not present Satl client, after image is build it is immediately available to the Saltboot clients.

Reviewer 2: It isn't completely clear to me what this statement (the first half of the sentence) actually means.  Maybe, a suggestion I made caused the trouble.

Author: I removed first sentence and focus on what is important: build image is immediately available to the clients without `image-sync` application.

This may have implications on how images are deployed to production.

> The following sections difference between containerized and regular workflow.

Reviewer 2: ```suggestion
The following sections differentiate between containerized and regular workflow.
```
